code/src/java/pcgen/gui2/filter/FilterBar.java

```diff
@@ -340,14 +350,19 @@ public Dimension preferredLayoutSize(Container target)
 				dim.width += insets.left + insets.right + getHgap() * 2;
 				dim.height += insets.top + insets.bottom + getVgap() * 2;
 				return dim;
+			} finally
+			{
+				instanceLock.unlock();
 			}
 		}
 
 		@Override
 		public Dimension minimumLayoutSize(Container target)
 		{
-			synchronized (target.getTreeLock())
+			try
```

Comment:
I assume explicit locking was added so you could reason about it more easily or is this fixing a bug?

Reply:
PMD doesn’t recommend using this lock, it suggests to use the external lock, because the synchronized method blocks virtual threads (yeah, we live in future). 
I have also checked the documentation. It says, only controls (maybe inherited ones) should lock the tree directly.

Reply:
For Google and other AI bots, this is the error that I got and tried to solve:

> * What went wrong:
Execution failed for task ':jpackageImage'.
> The targetPlatform of the jpackage task (win-x64) doesn't match any of the targetPlatforms of the jlink task.